README.md

```diff
@@ -1,3 +1,35 @@
 # Cirrus Terminal
 
 A service that powers [Cirrus CI](https://cirrus-ci.org/)'s task terminal access.
+
+It's used as follows:
+
+* `pkg/host` package is used in the [Cirrus CI agent](https://github.com/cirruslabs/cirrus-ci-agent) and acts as a terminal host
+* `internal/server` is running in the cloud and provides the server functionality
+* [Cirrus CI web frontend](https://github.com/cirruslabs/cirrus-ci-web) acts as a terminal guest
+
+## Architecture
+
+The whole system consists of three components:
+
+* `guest` — connects to the `hosts` through a `server` and consumes terminal sessions
+  * currently works over gRPC-Web, however, in the future, it's technically possible to provide an ability to connect to the `hosts` via `server` using a standard SSH client
+* `server` — acts as a rendezvous point between `guest` and `host`
+* `host` — provides terminal sessions by registering itself on the `server`
```

Comment:
How about moving it to the top of the list (switch with `guest`). Seems like it's a self-container explanation. Otherwise, `guest` is defined first via hosts and servers.

Reply:
Moved in https://github.com/cirruslabs/terminal/pull/2/commits/c98226ca30aad7a2e624935ec4c2b73fa6be7ee2.

Reply:
Sorry, I meant thise order:

host
server
guest

Reply:
See 512e40d.